server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -369,13 +397,13 @@ public void onAfter() {
         }
     }
 
-    private void onNoLongerPrimary(Exception failure) {
+    private void onNoLongerPrimary(Exception failure, ActionListener<Void> pendingActionlistener) {
```

Comment:
Maybe just named titled `listener` to more differentiate between this being local scoped and the field name.